pkg/metrics/backend_probe.go

```diff
@@ -0,0 +1,46 @@
+package mpcmetrics
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+)
+
+var errorThreshold = 0.5
+
+type BackendProbe struct {
+	successes atomic.Int32
+	failures  atomic.Int32
+}
+
+func NewBackendProbe() AvailabilityProbe {
+	return &BackendProbe{
+		successes: atomic.Int32{},
+		failures:  atomic.Int32{},
+	}
+}
+
+func (q *BackendProbe) CheckAvailability(_ context.Context) error {
```

Comment:
I wonder if we can emit simpler metrics and do this calculation on the Prometheus side with recording rules

Reply:
It is possible theoretically, but that might be not so reliable, as the transformer might become complex and need tweeks if some metrics logic is changed (or supplying unreliable data). So i decided to count that explicitly. And that might also allow us to add more health checks whih are not covered by regular metric counters. I did follo the general recoomendations here:

Reply:
It might indeed be useful to bring @yftacherzog and @amisstea into this discussion....

WDYT guys? Should we do this kind of processing in the code or should we use recording rules?

Reply:
I agree that it mainly depends on the sort of data that needs to be manipulated.

Reply:
ok, this discussion isn't leading anywhere useful it seems so I'm gonna close it.